### PR TITLE
Resolve xframe options warning

### PR DESCRIPTION
**Pull request by gherceg**
[Resolve security warnings](https://dimagi-dev.atlassian.net/browse/SAAS-11207)

##### SUMMARY

This addresses resolving one of the warnings from the ticket above.

My understanding is the `@xframe_options_sameorigin` decorator applies to commcare views that we intend to load inside of a frame, not views that contain a frame. 

The ability for 3rd party views to load inside of a frame are dictated by the 3rd party developers. So for example, our site contains an iframe that loads a page from google tag manager. Those developers are responsible for allowing that page to load inside of a frame regardless of the origin of the request.

In our case we need to apply the decorator to views that we know are loaded inside of a frame, but only when the request originates from our site. The default behavior for pages will be to never allow loading inside of a frame, regardless of origin.

##### RISK ASSESSMENT / QA PLAN

Yes. [Ticket here](https://dimagi-dev.atlassian.net/browse/QA-2069)

### CODE REVIEW

File: corehq/apps/hqwebapp/views.py [hunk 1]
```diff
@@ -468,6 +469,7 @@ def domain_login(req, domain, custom_template_name=None, extra_context=None):
     return _login(req, domain, custom_template_name, extra_context)
 
 
+@xframe_options_sameorigin
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/views.py:472]: This should probably also be applied tp `iframe_domain_login_new_window`

**Comment by orangejenny** [comment]
I grepped around for iframes and remembered that Web Apps uses them to display graphs. Do those need any adjustment?
Analytics also look like they use iframes, for google tag manager and ScheduleOnce. Will those areas still work as expected?

**Comment by gherceg** [comment]
> I grepped around for iframes and remembered that Web Apps uses them to display graphs. Do those need any adjustment?
> Analytics also look like they use iframes, for google tag manager and ScheduleOnce. Will those areas still work as expected?

@orangejenny My understanding is the `@xframe_options_sameorigin` decorator applies to _commcare views_ that we intend to load inside of a frame, not views that contain a frame. Additionally, the ability for 3rd party views to load inside of a frame are dictated by the 3rd party developers. So for example, google tag manager developers are responsible for allowing that page to load inside of a frame regardless of the origin of the request.

In our case we need to apply the decorator to views that we know are loaded inside of a frame, but only when the request originates from our site. Let me know if that doesn't clear it up, and if there are areas you think this decorator needs to be applied to. I'll update the PR description with this explanation. 

@dannyroberts helped clear this up for me originally so maybe he can add to/correct what I stated above.

**Comment by orangejenny** [comment]
@gherceg That makes sense, thank you for explaining - I think I was thinking about this inside out, if that makes sense.

File: corehq/apps/cloudcare/views.py [hunk 2]
```diff
@@ -289,6 +290,7 @@ class PreviewAppView(TemplateView):
     template_name = 'preview_app/base.html'
     urlname = 'preview_app'
 
+    @xframe_options_sameorigin
```
[the remark this answers is not in the document]
Comment (gherceg) [reply corehq/apps/cloudcare/views.py:293]: @mjriley I talked with Danny and the conclusion was while setting `X_FRAME_OPTIONS` to `DENY` may not make us much more secure than the default `SAMEORIGIN`, it's easy enough to do. The majority of overhead caused by this change has already been dealt with on this ticket by finding the spots that currently need to allow loading within a frame. In the future if a dev adds a view that needs to load within a frame, it _should_ be easy to detect why the frame is not able to load.

[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/cloudcare/views.py:293]: My concern isn't with the overhead to perform the change -- it's with the overhead to maintain the change. If a developer is tasked with modifying those current decorated views, they now have to look into the extra decorator. I don't like that developers now have more to understand and process for a feature that, as far as I can tell, doesn't target our use-case. That said, the conceptual overhead is still small, so if you're comfortable with the prior discussion, I'll table this.

[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/cloudcare/views.py:293]: Talked this over with Danny, and he's convinced me that implemented the changes you outlined are the correct path forward.

**Review by dannyroberts — APPROVED** [review]

**Comment by dannyroberts** [comment]
@gherceg what's the testing plan here? I put an "Awaiting QA" label just because I noticed there weren't any QA labels and this feature should for sure receive QA on staging (focusing particularly on web apps and app preview) before going out. If that's already happened then just replace the label with QA Passed

**Comment by gherceg** [comment]
@dannyroberts didn't realize this should go through QA so I'll submit a request now.

**Comment by gherceg** [comment]
This has made it through QA and is ready to go. See [here](https://dimagi-dev.atlassian.net/browse/QA-2069).

**Pull request merged**